Themis/ViewModels/Assessment/CodeEditorViewModel.swift

```diff
@@ -78,29 +93,42 @@ class CodeEditorViewModel: ObservableObject {
         }
     }
     
+    @MainActor
     func addInlineHighlight(feedbackId: UUID) {
         if let file = selectedFile, let selectedSection = selectedSection {
-            if (inlineHighlights.contains { $0.key == file.path }) {
-                inlineHighlights[file.path]?.append(
-                    HighlightedRange(
-                        id: feedbackId.uuidString,
-                        range: selectedSection,
-                        color: UIColor.systemYellow,
-                        cornerRadius: 8
-                    )
-                )
-            } else {
-                inlineHighlights[file.path] = [
-                    HighlightedRange(
-                        id: feedbackId.uuidString,
-                        range: selectedSection,
-                        color: UIColor.systemYellow,
-                        cornerRadius: 8
-                    )
-                ]
+            appendHighlight(feedbackId: feedbackId, range: selectedSection, path: file.path)
+        }
+            undoManager.endUndoGrouping() /// undo group with addFeedback in AssessmentResult
+    }
+    
+    @MainActor
+    func loadInlineHighlight(assessmentResult: AssessmentResult, participationId: Int) async {
+        for i in 0..<assessmentResult.inlineFeedback.count {
+            // extract file path
+            let feedback = assessmentResult.inlineFeedback[i]
```

Comment:
you could use for `feedback in assessmentResult.inlineFeedback` for the sake of readability